docs/source/architecture.data.md

```diff
@@ -0,0 +1,44 @@
+# Data Architecture
+
+Documentation covering data architecture for pyctyominer-transform.
+
+## Sources
+
+Data sources for pyctyominer-transform are measurement data created from other cell biology image analysis tools.
+
+See below for a brief overview of these sources and data types.
+
+- [CellProfiler](https://github.com/CellProfiler/CellProfiler) generates [CSV](https://en.wikipedia.org/wiki/Comma-separated_values) and [SQLite](https://www.sqlite.org/) databases data.
+- [Cytominer-database](https://github.com/cytomining/cytominer-database) generates [SQLite](https://www.sqlite.org/) databases (which includes table data based on CellProfiler CSV's mentioned above).
+- [DeepProfiler](https://github.com/cytomining/DeepProfiler) generates [NPZ](https://numpy.org/doc/stable/reference/routines.io.html?highlight=npz%20format#numpy-binary-files-npy-npz) data.
+
+## Data structure
+
+### Compartments and Images
+
+Data are organized into tables of generally two categories:
+
+- __Images__: Image tables generally include data about the image itself, including metadata.
+- __Compartments__: Compartment data tables (such as Cytoplasm, Cells, Nuclei, Actin, or others) includes measurements specific to an aspect or part of cells found within an image.
+
+### Identifying or Key Values
+
+Identifying or key values for image and compartment tables may include the following:
+
+- __ImageNumber__: Provides specificity on what image is being referenced (there may be many).
+- __ObjectNumber__: Provides specificity for a specific compartment object within an ImageNumber.
+- __Parent_Cells__: Provides a related Cell compartment ObjectNumber.
+- __Parent_Nuclei__: Provides a related Nuclei compartment ObjectNumber.
+
+### Image Data Relationships
+
+```{mermaid}
```

Comment:
Thanks for calling this out. The curly-brace notation `{mermaid}` instead of `mermaid` here is used by [sphinxcontrib-mermaid](https://github.com/mgaitan/sphinxcontrib-mermaid#markdown-support) to render within the built sphinx documentation site content by way of [myst-parser](https://myst-parser.readthedocs.io/en/latest/). The rendered content can be previewed using `dagger do docs` or `poetry run sphinx-build docs/source docs/build`. In the future I hope to find a way to retain Github's rendering of this content without a build (maybe this is worth an issue?).

Reply:
oh interesting - most people will see this in the RTD, right? So probably nothing to followup with here

Reply:
@gwaybio thanks, and correct - that's the intent at least (most people should see this in the rendered html docs) 🙂. For now, I'll mark this as resolved.

Aside: It could really improve things if we were able to see the rendered output within Github as-is, and it feels like we're close to that point via those libraries. To this effect, I created [an issue](https://github.com/mgaitan/sphinxcontrib-mermaid/issues/99) to see if we can enable this functionality in the future via Github-compatible markdown syntax.